packages/table-react/src/Table.tsx

```diff
@@ -1,31 +1,92 @@
-import React from "react";
-import { TableAnchorRowData, TableRow } from "./TableRow";
+import React, { ReactNode } from "react";
+import { TableRowAnchor } from "./TableRowAnchor";
+import { TableRowAccordion } from "./TableRowAccordion";
+import { TableRowSimple } from "./TableRowSimple";
 
-export type TableRowData = string[] | TableAnchorRowData;
+//-------TODO:  Skulle vi ha flyttet alle disse interfacene, typene (og type guardsene?) ut i en egen fil? -------------
+export type TableRowData = ReactNode[] | TableRowAnchorData | TableRowAccordionData; // Endret fra string. Greit?
+
+export type TableRowDataVariants = "anchor" | "accordion";
+
+export interface BaseTableRowData {
+    type: TableRowDataVariants;
+}
+
+export interface TableRowAnchorData extends BaseTableRowData {
+    rowData: string[];
+    type: "anchor";
+    href: string;
+    hrefLabel: string;
+    onRowClick?: (href: string) => void;
+}
+
+export interface TableRowAccordionData extends BaseTableRowData {
+    rowData: string[];
+    type: "accordion";
+    defaultOpen: boolean;
+    children: ReactNode[];
+}
+
+export function isAnchorRowData(
+    row: ReactNode[] | TableRowAnchorData | TableRowAccordionData,
+): row is TableRowAnchorData {
+    // Endret fra string. Greit?
+    return "type" in row && row.type === "anchor";
+}
+
+export function isAccordionRowData(
+    row: ReactNode[] | TableRowAnchorData | TableRowAccordionData,
+): row is TableRowAccordionData {
+    // Endret fra string. Greit?
+    return "type" in row && row.type === "accordion";
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
 
 interface Props {
     columns: string[];
     rows: Array<TableRowData>;
     className?: string;
+    accordion?: boolean;
 }
 
-export function Table(props: Props) {
+export function Table({ columns, rows, className, accordion }: Props) {
+    const componentClassName = "jkl-table".concat(
+        accordion ? ` jkl-table--accordion` : "",
+        className ? ` ${className}` : "",
+    );
+
+    const createAppropriateTableRow = (row: TableRowData, index: number) => {
+        // TODO: Good/bad practice å ha keys her?
```

Comment:
No need for key here, inside this scope there is no reorder reconcilliation conflict. The key should go where the map is. And index is not too good of a solution. Should push this out to client developer to supply a uniqe key for each row/col.